e2e/e2e_test.go

```diff
@@ -198,6 +198,28 @@ func TestGetAndPutE2E(t *testing.T) {
 			expectedCommitCount: 9,
 			expectedCommits:     []string{"Push 2."},
 		},
+		{
+			description: "get works when checkout",
+			source: resource.Source{
+				Repository:  "itsdalmo/test-repository",
+				V3Endpoint:  "https://api.github.com/",
+				V4Endpoint:  "https://api.github.com/graphql",
+				AccessToken: os.Getenv("GITHUB_ACCESS_TOKEN"),
+			},
+			version: resource.Version{
+				PR:            targetPullRequestID,
+				Commit:        targetCommitID,
+				CommittedDate: time.Time{},
+			},
+			getParameters: resource.GetParameters{
+				IntegrationTool: "checkout",
+			},
+			putParameters:       resource.PutParameters{},
+			versionString:       `{"pr":"4","commit":"a5114f6ab89f4b736655642a11e8d15ce363d882","committed":"0001-01-01T00:00:00Z"}`,
+			metadataString:      `[{"name":"pr","value":"4"},{"name":"url","value":"https://github.com/itsdalmo/test-repository/pull/4"},{"name":"head_name","value":"my_second_pull"},{"name":"head_sha","value":"a5114f6ab89f4b736655642a11e8d15ce363d882"},{"name":"base_name","value":"master"},{"name":"base_sha","value":"93eeeedb8a16e6662062d1eca5655108977cc59a"},{"name":"message","value":"Push 2."},{"name":"author","value":"itsdalmo"}]`,
+			expectedCommitCount: 7,
+			expectedCommits:     []string{"Push 2."},
```

Comment:
I think since this is the only E2E test that does a plain `checkout`, I think it might be a good idea the include the entire git history here to verify the functionality. This is what the history should look like:

```go
			expectedCommits: []string{
				"Push 2.",
				"Push 1.",
				"Add another commit to the 2nd PR to verify concourse behaviour.",
				"Add another comment to 2nd pull request.",
				"Add comment from 2nd pull request.",
				"Add comment after creating first pull request.",
				"Initial commit",
			},
```

Reply:
ah, good point!